lib/spack/spack/test/build_systems.py

```diff
@@ -320,6 +320,10 @@ def test_cmake_std_args_hip(self, default_mock_concretization):
         option = spack.build_systems.cmake.CMakeBuilder.define_hip_architectures(s.package)
         assert "-DCMAKE_HIP_ARCHITECTURES:STRING=gfx900" == option
 
+    def test_cmake_dependent_args(self, concretize_and_setup):
+        s = concretize_and_setup("cmake-client-dep")
+        assert s.package.builder.dependency_args == ["-DDEP_TEST_ARG=ON"]
```

Comment:
Perhaps a slightly-more robust test would mock `.cmake()` itself, such that when the build calls it, you can check that it includes the extra args (this would make sure that `.dependency_args` is mixed in as needed).

Reply:
Sorry `CMakeBuilder.cmake` calls `pkg.module.cmake` after assembling the options, so I was hoping you could mock the latter?